save-backend/src/main/kotlin/com/saveourtool/save/backend/service/OrganizationService.kt

```diff
@@ -46,22 +46,89 @@ class OrganizationService(
      * @return deleted organization
      */
     @Suppress("UnsafeCallOnNullableType")
-    fun deleteOrganization(organizationName: String): Organization = getByName(organizationName)
+    fun deleteOrganization(organizationName: String): Organization =
+            if (!hasProjects(organizationName)) {
+                changeOrganizationStatus(organizationName, OrganizationStatus.DELETED)
+            } else {
+                getByName(organizationName)
+            }
+
+    /**
+     * Mark organization with [organizationName] and all it`s projects as banned
+     *
+     * @param organizationName an [Organization]'s name to banned
+     * @return deleted organization
+     */
+    fun banOrganization(organizationName: String): Organization {
+        val projects = projectService.getAllByOrganizationNameAndStatus(organizationName)
+        projects.forEach {
+            it.status = ProjectStatus.BANNED
+            projectService.updateProject(it)
```

Comment:
it's two different operations: on projects and on organization. We need to make them in a single transaction